src/insights_client/tests/test_metrics.py

```diff
@@ -285,3 +285,115 @@ def test_metrics_post_event_proxy(post, config_file_factory, rhsm_config_file_fa
         proxies={"https": "http://user:password@localhost:3128"},
     )
 
+@mark.parametrize(("config", "delenv", "setenv", "sysargv", "is_offline"), (
+    ("", ("INSIGHTS_OFFLINE"), (), ["insights-client"], False),
+    ("offline=True\n", ("INSIGHTS_OFFLINE"), (), ["insights-client"], True),
+    ("", ("INSIGHTS_OFFLINE"), (), ["insights-client", "--offline"], True),
+    ("", ("INSIGHTS_OFFLINE"), (("INSIGHTS_OFFLINE", "True"),), ["insights-client"], True),
+))
```

Comment:
```suggestion
    ("offline=True\n", (), (("INSIGHTS_OFFLINE", "False"),), ["insights-client"], False),
    ("offline=False\n", (), (("INSIGHTS_OFFLINE", "False"),), ["insights-client", "--offline"], True),
))
```

- Verify that it is possible to override the config file by the environment value.
- Verify that the command line argument trumps all.